sig-architecture/production-readiness.md

```diff
@@ -28,6 +27,19 @@ aspects of the process.
     happens if it is subsequently upgraded again?
   - Are there tests for this?
 * Scalability
+  - Does enabling/using the feature will result in new API calls?
```

Comment:
This is a _really_ broad phrasing of this question, and I'm not sure there's a lot of value in enumerating every API call made by a controller related to a feature. It might be more helpful to be more specific about the types of calls we know are problematic:
* Does this feature require something running on nodes (kubelets, kube-proxy, daemonset pod, etc) to list or watch all instances of a resource (with or without a labelSelector/fieldSelector filter)?
* Does this feature require a component to continuously update or process instances of a resource (e.g. the cronjobs controller polling implementation, or the kubelet node heartbeat implementation, leader election, etc) or can components reach a steady state in which they do no work if no API changes are observed?

Reply:
I'm not sure I really agree. Every reactive API call is also something that we should catch (e.g. if I send 3 API calls, for every change of pod (no matter if this is kubelet or other controller), this is also something we really want to catch).

Being so broad in the question can actually let us catch classes of issues we didn't explicitly think through when creating this question.

[Maybe we should formulate it a bit differently, but I actually would really like to understand all new API calls a feature is adding.]

Reply:
> I actually would really like to understand all new API calls a feature is adding

My prediction is that a question that broad will not be answered as exhaustively as you hope

Reply:
I got your point. I wasn't trying to say that I believe it is perfect, I was just trying to say that I think that if we will be too specific, we will miss some cases for sure.

I'm going to think a bit more how to make improve it.

Reply:
Rephrased to make the question more focused on that. Hopefully it's better now.